addons/mail/static/src/owl/components/followers_menu/followers_menu.js

```diff
@@ -0,0 +1,143 @@
+    constructor(...args) {
+        super(...args);
+        this.state = useState({
+            hasDropdown: false,
+        });
+        this.storeDispatch = useDispatch();
+        this.storeProps = useStore((state, props) => {
+            const thread = state.threads[props.threadLocalId];
+            const followerLocalIds = thread ? thread.followerLocalIds : [];
+            const followers = followerLocalIds.map(
+                followerLocalId => state.followers[followerLocalId]);
+            const isFollowing = !!followers.find(
+                follower => follower.partnerId === this.env.session.partner_id);
+            return {
+                followersAmount: followerLocalIds.length,
+                isFollowing,
+                thread,
+            };
+        });
+        this._dropdownRef = useRef('dropdown');
+        this._onClickCaptureGlobal = this._onClickCaptureGlobal.bind(this);
+    }
+
+    /**
+     * @override
+     */
+    mounted() {
+        document.addEventListener('click', this._onClickCaptureGlobal, true);
+    }
+
+    /**
+     * @override
+     */
+    willUnmount() {
+        document.removeEventListener('click', this._onClickCaptureGlobal, true);
+    }
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    _hide() {
+        this.state.hasDropdown = false;
+    }
+
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    async _onClickAddChannels(ev) {
+        ev.preventDefault();
+        this._hide();
+        await this.storeDispatch('addChannelFollowersToThread', this.props.threadLocalId);
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    async _onClickAddFollowers(ev) {
+        ev.preventDefault();
+        this._hide();
+        await this.storeDispatch('addPartnerFollowersToThread', this.props.threadLocalId);
+    }
+
+    /**
+     * Close the dropdown when clicking outside of it.
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickCaptureGlobal(ev) {
+        // since dropdown is conditionally shown based on state, dropdownRef can be null
+        if (this._dropdownRef.el && !this._dropdownRef.el.contains(ev.target)) {
+            this._hide();
+        }
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickFollow(ev) {
+        this.storeDispatch('followThread', this.props.threadLocalId);
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickFollowersButton(ev) {
+        this.state.hasDropdown = !this.state.hasDropdown;
+    }
+
+    /**
+     * @private
+     * @param {MouseEvent} ev
+     */
+    _onClickFollower(ev) {
+        this._hide();
```

Comment:
A follower contains 3 clickable elements : edit, delete and the name of the partner/channel. Whether you click on one or another, the dropdown should be closed otherwise the opened dialog just hide the form view but you can see that the dropdown is opened behind

Reply:
I think it makes more sense to extend the click on name to click on whole follower item

Reply:
What do you mean by extend ?

Reply:
I think `Follower._onClick()` should replace `Follower. _onClickDetails()`.

Another valid solution IMO:
- keep click on details, but it should not close the dropdown menu
- add a visual style feedback on hover of follower details, so it's clear there is some interaction with a click there.

Reply:
With both or your solutions, I can't see how you can manage to close the dropdown when you click on edit or remove ? In fact, details and remove do not "really need" to close the dropdown (as the user will be redirected or the follower deleted from the dropdown) but edit do. So edit should trigger up an event in these solutions ?

With the first solution, if you click on edit, won't the general follower click be triggered and so the redirect action executed whereas we don't want to ?

Concerning visual feedback, you mean something more visible than the slight color change actually present ?

Reply:
Don't forget there's `ev.stopPropagation()` to prevent handling by parented nodes through bubbling.

Reply:
I really don't see the improvement ... but I tried to change something in a way that could be what you want... If it's not ok then, maybe you could make a commit directly on it so I can see what you mean ?